### PR TITLE
Refactor StartDevMode to return a single object instead of multiple args

### LINKED ISSUE #6876: Doc automation tests broken due to new NodeJS Stacks versions in staging registry
/kind bug
/area testing

## Actual behavior

```
  [FAILED] registry_list_output.mdx
  Expected
      <string>:   strings.Join({
        	... // 108 identical bytes
        	"                        VERSIONS\nnodejs-react  StagingRegistry  ",
        	"       React is a free and open-source front-en...  2.0.2, 2.1.0",
      + 	", 2.2.0",
        	"\nnodejs-react  DefaultDevfileRegistry  React is a free and open-",
        	"source front-en...  2.0.2, 2.1.0\n```\n",
        }, "")
      
  to be empty
  In [It] at: /go/odo_1/tests/documentation/command-reference/doc_command_reference_init_test.go:189 @ 06/09/23 12:49:03.116
------------------------------
â€¢â€¢â€¢â€¢â€¢â€¢Spawning '/go/bin/odo init' from /tmp/591808707
â€¢â€¢â€¢Spawning '/go/bin/odo delete project odo-dev' from /tmp/2208322380
â€¢â€¢â€¢â€¢

Summarizing 1 Failure:
  [FAIL] doc command reference odo init Non Interactive Mode fetching devfile from a registry when setting up the registry [It] Fetch Devfile from any registry of the list [nocluster]
  /go/odo_1/tests/documentation/command-reference/doc_command_reference_init_test.go:189

Ran 16 of 16 Specs in 46.762 seconds
FAIL! -- 15 Passed | 1 Failed | 0 Pending | 0 Skipped
```

New versions of the `nodejs-*` stacks have been added in https://github.com/devfile/registry/pull/187

## Expected behavior

Like in the integration tests, doc tests should rely on our internal Devfile registry instead.

## Any logs, error output, etc?

Issue noticed in several PRs:
- #6872 (https://s3.eu-de.cloud-object-storage.appdomain.cloud/odo-tests-openshift-logs/pr-6872-kubernetes-docs-tests-805.txt)
- #6875 (https://s3.eu-de.cloud-object-storage.appdomain.cloud/odo-tests-openshift-logs/pr-6875-kubernetes-docs-tests-798.txt)
- #6871 (https://s3.eu-de.cloud-object-storage.appdomain.cloud/odo-tests-openshift-logs/pr-6871-kubernetes-docs-tests-795.txt)

### PR DESCRIPTION
**Pull request by valaparthvi**
**What type of PR is this:**
/area testing
/kind code-refactoring

**What does this PR do / why we need it:**
I find it confusing to work with multiple args returned by `StartDevMode`, using a single object makes it easier.

**Which issue(s) this PR fixes:**

Latest commit Fixes #6876 

**PR acceptance criteria:**

- [ ] Unit test 

- [x] Integration test 

- [ ] Documentation 

**How to test changes / Special notes to the reviewer:**

### CODE REVIEW

File: tests/helper/helper_dev.go [hunk 1]
```diff
@@ -314,13 +332,17 @@ func WaitForDevModeToContain(options DevSessionOpts, substring string, stopSessi
...
 	}
 	err = session.Err.Clear()
 	if err != nil {
-		return DevSession{}, nil, nil, err
+		return DevSessionOutput{Err: err}
+	}
+	return DevSessionOutput{
+		Session: result,
+		StdOut:  string(outContents),
+		ErrOut:  string(errContents),
```
Comment (valaparthvi) [inline tests/helper/helper_dev.go:344]:
```suggestion
		ErrOut:  string(errContents),
		Endpoints: getPorts(string(outContents), options.CustomAddress),
```

**Review by feloy — APPROVED** [review]
Thanks @valaparthvi for this improvement

**Comment by valaparthvi** [comment]
/override OpenShift-Integration-tests/OpenShift-Integration-tests
Network issue.
```sh
  [odo]  âœ—  Syncing files into the container [4ms]
  [odo] Error occurred on Push - watch command was unable to push component: failed to sync to component with name mynode: failed to sync to component with name mynode: unable push files to pod: error while streaming command: error sending request: Post "https://c100-e.eu-de.containers.cloud.ibm.com:31049/api/v1/namespaces/cmd-remove-binding-test71qon/pods/mynode-app-57d77b45bb-xmjvz/exec?command=tar&command=xf&command=-&command=-C&command=%2Fprojects&command=--no-same-owner&container=runtime&stderr=true&stdin=true&stdout=true": dial tcp: lookup c100-e.eu-de.containers.cloud.ibm.com: operation was canceled
  [odo] 
  [odo] 
  [odo] â†ª Dev mode
  [odo]  Status:
  [odo]  Watching for changes in the current directory /tmp/3271381815
  [odo] 
  [odo]  Keyboard Commands:
  [odo] [Ctrl+c] - Exit and delete resources from the cluster
  [odo]      [p] - Manually apply local changes to the application on the cluster
  [odo] I0616 11:43:50.565792   17461 implem.go:114] The path for preference file is /tmp/4157871543/preference.yaml
  [odo] I0616 11:43:50.566355   17461 segment.go:264] Checking telemetry enable status
  [odo] I0616 11:43:50.566388   17461 segment.go:282] Sending telemetry disabled by env variable
  [odo] Cleaning resources, please wait
  [odo] I0616 11:43:50.614310   17461 delete.go:176] Ignoring resource: servicebindings/my-nodejs-app-cluster-sample-ocp; it does not exist on the cluster
  [odo] I0616 11:43:50.657616   17461 all.go:46] starting to concurrently query 200 APIs
  [odo] I0616 11:43:50.659814   17461 all.go:62] fired up all goroutines to query APIs
  [odo] I0616 11:43:51.528455   17461 all.go:67] all goroutines have returned in 870.837356ms
  [odo] I0616 11:43:51.528485   17461 all.go:76] query result: objects=546
  [odo]  âœ—  Dev mode interrupted by user
  Deleting project: cmd-remove-binding-test71qon
  Running oc with args [oc delete project cmd-remove-binding-test71qon --wait=false] and odo env: []
  [oc] project.project.openshift.io "cmd-remove-binding-test71qon" deleted
  Setting current dir to: /go/odo_1/tests/integration
  Deleting dir: /tmp/3271381815
  Deleting dir: /tmp/4157871543
  << Timeline

  [FAILED] Expected
      <string>: 
  to contain substring
      <string>: not found
```

**Comment by valaparthvi** [comment]
/override windows-integration-test/Windows-test
```sh
  [FAILED] No future change is possible.  Bailing out early after 1.212s.
  Running oc.exe with args [oc get csv -o jsonpath={.items[?(@.status.phase=="Succeeded")].metadata.name}] and odo env: []
  Expected
      <int>: 1
  to match exit code:
      <int>: 0
------------
  I0616 05:03:50.613347    6492 util.go:198] path C:\Users\Administrator.ANSIBLE-T
  EST-VS\AppData\Local\Temp\2170439783\devfile.yaml doesn't exist, skipping it    [92m
  ? [97mDo you want to list services from:[m  [36m[Use arrows to move, type to filter][m      [36m
  > current namespace[m[61X[61C
    all accessible namespaces[53X[53C[?25h[?25l[HI[?25h
  ======================
  Unexpected error:
      <*xpty.errPassthroughTimeout | 0xc0008cc020>: {
          error: <*errors.errorString | 0xc0008cc010>{
              s: "passthrough i/o timeout",
          },
      }
      passthrough i/o timeout
  occurred
  In [It] at: C:/Users/Administrator.ANSIBLE-TEST-VS/4170/tests/helper/helper_interactive.go:111 @ 06/16/23 05:05:50.661
------------------------------
Summarizing 3 Failures:
  [FAIL] odo devfile deploy command tests deploying devfile with long-running exec when Automount volumes are present in the namespace [It] should mount the volumes
  C:/Users/Administrator.ANSIBLE-TEST-VS/4170/tests/helper/helper_cmd_wrapper.go:120
  [FAIL] odo add binding interactive command tests when running a deployment when binding to a service in a different namespace [It] should successfully add binding without devfile
  C:/Users/Administrator.ANSIBLE-TEST-VS/4170/tests/helper/helper_interactive.go:111
  [FAIL] odo describe/list binding command tests when creating a component with a binding as environment variables (service in namespace "") when Starting a Pg service [BeforeEach] when running dev session when changing the current namespace should list the binding with --namespace flag - command: [list binding]
  C:/Users/Administrator.ANSIBLE-TEST-VS/4170/tests/helper/helper_generic.go:58

```
```

**Comment by valaparthvi** [comment]
/override Pull Request Website Preview / deploy-prev

**Pull request merged**